### PR TITLE
Bump on py-transformer-engine versions

### PR DESCRIPTION
**Pull request by rfbgo**

### CODE REVIEW

**Review by aurianer — COMMENTED** [review]
Hi, thanks for this, I was preparing a PR for that, would you mind updating the required dependencies of each versions? For example install requirements of 1.7 are listed [here](https://github.com/NVIDIA/TransformerEngine/blob/release_v1.7/setup.py#L246-L250) and [here](https://github.com/NVIDIA/TransformerEngine/blob/release_v1.7/setup.py#L269). Setup requirements are listed [here](https://github.com/NVIDIA/TransformerEngine/blob/release_v1.7/setup.py#L263) and [here](https://github.com/NVIDIA/TransformerEngine/blob/release_v1.7/setup.py#L265). Same check applies for other versions if you need them, if you don't you can just add 1.7 :)
I can also push my changes on your PR if you want

File: var/spack/repos/builtin/packages/py-transformer-engine/package.py [hunk 1]
```diff
@@ -28,6 +31,7 @@ class PyTransformerEngine(PythonPackage):
     depends_on("cmake@3.18:")
     depends_on("py-pydantic")
     depends_on("py-importlib-metadata")
+    depends_on("gdrcopy")
```
Comment (aurianer) [inline var/spack/repos/builtin/packages/py-transformer-engine/package.py:34]: Is this necessary, it doesn't seem to be listed in the setup.py of TE

Comment (rfbgo) [reply var/spack/repos/builtin/packages/py-transformer-engine/package.py:34]: Honestly, my work on this package was quite a while ago and I don't remember so clearly. Perhaps it is best if I abandon my PR in favor of yours

My recollection was that I was getting a missing header around `gdrapi.h` without it, but I agree looking through the package source I don't see why that would happen

Comment (aurianer) [reply var/spack/repos/builtin/packages/py-transformer-engine/package.py:34]: I opened this: #45190 but I still have one problem to fix installing thrift

Comment (rfbgo) [reply var/spack/repos/builtin/packages/py-transformer-engine/package.py:34]: Awesome! Thanks!

**Pull request closed without merging**